src/download.test.ts

```diff
@@ -41,7 +41,12 @@ describe("downloadFile", () => {
 
     const path = extractNonSkylinkPath(fullSkylink, skylink);
 
-    let fullExpectedUrl = `${expectedUrl}${path}${attachment}`;
+    let fullExpectedUrl;
+    if (path !== "") {
+      fullExpectedUrl = `${expectedUrl}/${path}${attachment}`;
+    } else {
+      fullExpectedUrl = `${expectedUrl}${attachment}`;
+    }
```

Comment:
I'm going to make it non blocking but I am strongly against including a code like that.
1. it's not readable as to what should be the result of the assertion
2. cannot compare the inputs and expected values when doing code review
3. running code inside a test of a different piece of code brings a potential of false positives - that code you run is not unit tested either

My take on it is that the expected values should not be generated. If that's too much of work to prepare the array by hand, just take that piece of code, map the inputs and console.log the results but then paste them in the test, basically hardcode them so we could glance at the inputs and expected outputs.

Reply:
I agree that writing the tests with `extractNonSkylinkPath` is not ideal, but I believe it is a practical and time-saving solution given the very large number of inputs. I do not think that spending an hour or more on improving this test (and others where we use `extractNonSkylinkPath`) is a productive use of developer time.

Reply:
It will not take an hour to do that, it's just writing a map function and console.log the result and copy&pasting the result as a variable. Anyway I'm not blocking on that, it can be a follow up fwiw.